components/Deck/ContentModulesPanel/CollectionsPanel/CollectionsPanel.js

```diff
@@ -0,0 +1,171 @@
+            },
+            removeDeckError: {
+                id: 'CollectionsPanel.error.removeDeck', 
+                defaultMessage: 'An error occured while removing playlist from deck...'
+            }, 
+            addDeckError: {
+                id: 'CollectionsPanel.error.adDeck', 
+                defaultMessage: 'An error occured while adding playlist to the deck...'                
+            }
+        });
+    }
+    showNewCollectionModal(event){
+        event.preventDefault();
+        this.setState({
+            showNewCollectionModal: true
+        });
+    }
+    showErrorPopup(text){
+        let title = this.context.intl.formatMessage(this.messages.errorTitle);
+        swal({
+            title: title,
+            text: text,
+            type: 'error',
+            timer: 2000,
+            showCloseButton: false,
+            showCancelButton: false,
+            allowEscapeKey: false,
+            showConfirmButton: false
+        })
+        .then(() => {/* Confirmed */}, (reason) => {/* Canceled */});
+    }
+    addCollection(deckId, event, data){
+        let collectionId = data.value;
+        let collectionDetails = {};
+
+        // transform collection details from array to json
+        collectionDetails = this.userCollections.reduce((details, value, key) => {
+            details[value._id] = value; return details;
+        }, {});
+
+        let collectionToAdd = collectionDetails[collectionId];
+        this.context.executeAction(addDeckToCollection, {
+            collectionId: collectionToAdd._id,
+            deckId: deckId,
+            collection: collectionToAdd,
+        });
+
+        this.setState({
+            currentSelection: ''
+        });
+    }
+    render() {
+        if (this.props.DeckCollectionStore.removeDeckFromCollectionError) {
+            this.showErrorPopup(this.context.intl.formatMessage(this.messages.removeDeckError));
+        } else if (this.props.DeckCollectionStore.addDeckToCollectionError) {
+            this.showErrorPopup(this.context.intl.formatMessage(this.messages.addDeckError));  
+        }
+
+        const userId = this.props.UserProfileStore.userid;
+        const selector = this.props.DeckCollectionStore.selector;
+        const groups = this.props.UserProfileStore.user.groups;
+        const groupIds = (groups || []).map( (group) => group.id);
+
+        // collections of the current deck
+        const deckCollections = this.props.DeckCollectionStore.deckCollections;
+
+        // collections that the user has edit rights
+        this.userCollections = [];
+        if (this.props.DeckCollectionStore.collections !== undefined) {
+            this.userCollections = this.props.DeckCollectionStore.collections.documents;
+        }
+        
+        // find collection dropdown options
+        let deckCollectionIds = deckCollections.map( (col) => col._id);
+        let collectionDropdownOptions = this.userCollections.filter( (collection) => {
```

Comment:
altering this .filter function might help to only show collections that the user is admin of, and can thus assign a deck to.

Reply:
Actually, you can select from playlists that either you are the admin or you belong to a user group that is assigned to that playlist. So, I think that in both cases you should have the right to add decks to that playlist.

Reply:
Ah ofcourse, I forgot about user groups :) 👍  . Would it then be also good to remove playlists from decks that you have user group access to? (I recall this is currently not possible)

Reply:
I am not sure what you mean by that. If you have edit rights to a deck you will also have the right to remove it from all playlists? no matter if you have edit rights on the playlist? Hmm, I am not sure if we want this.

But, if you mean that currently you cannot remove a deck from a playlist in the playlist's page despite the fact that you have edit right to that playlist, then you are right! I have included this enhancement in PR https://github.com/slidewiki/slidewiki-platform/pull/973